src/vm/eventpipeeventinstance.cpp

```diff
@@ -79,16 +79,35 @@ unsigned int EventPipeEventInstance::GetAlignedTotalSize() const
     CONTRACT_END;
 
     // Calculate the size of the total payload so that it can be written to the file.
-    unsigned int payloadLength =
-        sizeof(m_metadataId) +          // Metadata ID
-        sizeof(m_threadID) +            // Thread ID
-        sizeof(m_timeStamp) +           // TimeStamp
-        sizeof(m_activityId) +          // Activity ID
-        sizeof(m_relatedActivityId) +   // Related Activity ID
-        sizeof(m_dataLength) +          // Data payload length
-        m_dataLength +                  // Event payload data
-        sizeof(unsigned int) +          // Prepended stack payload size in bytes
-        m_stackContents.GetSize();      // Stack payload size
+    unsigned int payloadLength = 0;
+
+    if (format == EventPipeNetPerfFormatV3)
+    {
+        payloadLength =
+            sizeof(m_metadataId) +          // Metadata ID
+            sizeof(DWORD) +                 // Thread ID
+            sizeof(m_timeStamp) +           // TimeStamp
+            sizeof(m_activityId) +          // Activity ID
+            sizeof(m_relatedActivityId) +   // Related Activity ID
+            sizeof(m_dataLength) +          // Data payload length
+            m_dataLength +                  // Event payload data
+            sizeof(unsigned int) +          // Prepended stack payload size in bytes
+            m_stackContents.GetSize();      // Stack payload size
+    }
+    else if (format == EventPipeNetTraceFormatV4)
+    {
+        payloadLength =
+            sizeof(m_metadataId) +          // Metadata ID
+            sizeof(m_threadId) +            // Thread ID
+            sizeof(m_timeStamp) +           // TimeStamp
+            sizeof(m_activityId) +          // Activity ID
+            sizeof(m_relatedActivityId) +   // Related Activity ID
+            sizeof(m_dataLength) +          // Data payload length
+            m_dataLength +                  // Event payload data
+            sizeof(unsigned int) +          // Prepended stack payload size in bytes
+            m_stackContents.GetSize();      // Stack payload size
+    }
```

Comment:
Maybe it should be like this:
```cpp
    unsigned int threadIdSize = 0;

    if (format == EventPipeNetPerfFormatV3)
    {
        threadIdSize = sizeof(DWORD);
    }
    else if (format == EventPipeNetTraceFormatV4)
    {
        threadIdSize = sizeof(m_threadId);
    }
    else
    {
        // Error out?
    }

    payloadLength =
        sizeof(m_metadataId) +         // Metadata ID
        threadIdSize +                          // Thread ID
        sizeof(m_timeStamp) +          // TimeStamp
        sizeof(m_activityId) +             // Activity ID
        sizeof(m_relatedActivityId) + // Related Activity ID
        sizeof(m_dataLength) +          // Data payload length
        m_dataLength +                      // Event payload data
        sizeof(unsigned int) +             // Prepended stack payload size in bytes
        m_stackContents.GetSize();    // Stack payload size
```

Reply:
I added the error out part, but I left the sums separate because several more changes accumulate over the coming commits.